lib/jekyll/filters/url_filters.rb

```diff
@@ -23,9 +23,9 @@ def absolute_url(input)
       # Returns a URL relative to the domain root as a String.
       def relative_url(input)
         return if input.nil?
-        return ensure_leading_slash(input.to_s) if sanitized_baseurl.nil?
+        parts = [sanitized_baseurl, input]
```

Comment:
This method,[`sanitized_baseurl`](https://github.com/jekyll/jekyll/blob/b0136fccfab51fd7885207fa79bc3dbf3f64c6f4/lib/jekyll/filters/url_filters.rb#L44-L47), always returns a string, which means `parts.compact` is a no-op. This appears to invalidate our assumptions here?